include/astra_camera/astra_driver.h

```diff
@@ -69,11 +69,79 @@
 
 #include <ros/ros.h>
 
+#include <actionlib/server/simple_action_server.h>
+#include <camera_control_msgs/GrabImagesAction.h>
+
+#include <diagnostic_updater/diagnostic_updater.h>
+#include <diagnostic_updater/publisher.h>
+
 namespace astra_wrapper
 {
 
+struct CameraParameters
+{
+  int exposure;
+  int gain;
+  bool auto_exposure;
+};
+
+/**
+ * @brief The ActionImageSyncer struct
+ * @note When the change in a parameter of the camera is big, the camera is
+ * slow to react and won't give the desired image. Skipping frames accommodates
+ * for this. Also, keep in mind that the camera is queried about a parameter, it
+ * responds with the set value and not the currently reached one.
+ */
+struct ActionImageSyncer
+{
+  ActionImageSyncer() :
+      num_skip_frames_default(2),
+      frames_skipped(0),
+      need_image(false),
+      acquired(true)
+  {}
+
+  void reset(bool skip_frames)
+  {
+    num_skip_frames = skip_frames ? num_skip_frames_default : 0;
+    frames_skipped = 0;
+    need_image = true;
+    acquired = false;
+  }
+
+  void check()
+  {
+    if (frames_skipped == num_skip_frames)
+    {
+      need_image = false;
+      acquired = true;
+    }
+    else
+    {
+      ++frames_skipped;
+    }
+  }
+
+  bool ready()
+  {
+    return acquired;
+  }
+
+  boost::mutex mutex;
+  boost::condition_variable cv;
+  const int num_skip_frames_default;
```

Comment:
better go for unsigned values, they have a defined overflow behavior and are what you want here